src/jblockmain/IPlottable.java

```diff
@@ -12,5 +12,5 @@
     ArrayList<Double> getXCtPoints(int blockNumber) throws Exception;
     ArrayList<Double> getYCtPoints(int blockNumber) throws Exception;
     int getNumberOfBlocksToPlot();
-    void writeToDXF(File fileOutput);
+    void writeToDXF(File fileOutput, boolean[] dxfLayerChooser);
```

Comment:
Just a general comment but this is not great for an interface as there is no indication of the size of `boolean[]`. Although your choice of using a `boolean[]` to pass information is the simplest, it is not the best from a design perspective. We should have a discussion about how to do this better.

Reply:
The reason for that is that we may, in the future, add more layers. As of now I can't think of any but after the software gets tested by fashion technology academics they might come up with one or two more. Once we have a finished layer list I think it might be better to make it a plain array.

Reply:
The reason you have given is exactly the reason why this solution is bad.

Reply:
Interfaces are essentially contracts. They are classes without any implementation. They need to give sufficient information for anyone implementing them to be able to implement them properly and unambiguously. By adding an unknown-sized array to an interface definition there is high likelihood of the implementer having no idea how to use it.

Reply:
We can discuss better ways to implement this next week